### ISSUE TITLE
No notification on receiving messages

### ISSUE THREAD

**Issue by LakshyAAAgrawal** [vector-im/riot-android#2266]
Just to test it, i sent a message on Riot from a friends phone( which was a riot iOS ) to my phone while my own phone switched off. When I switched my phone on, there was no notification of any message on my phone. When I opened Riot, then it displayed unread message and also an Android notification.

I am using Google play version.

**Comment by geromueller** [comment]
Happens onmy phone as well. Using:
- googleplay version and currently 0.8.11-dev.
- AOSPExtended on Redmi4X with microG. All other apps work fine, including GCM.

Riot does not even register for Notifications! Uninstalled and cleared data and cache multiple times.

**Comment by eddie1030** [comment]
Hi

I just created a github account to confirm this behavior, as all users I have introduced riot to, complain about not receiving notifications after a while. I can provide data for all individual devices used, but I know it's a mix of android and iphone users.

I think a solid notification implementation is key for wider user adoption, as user stop using the app because they basically forget about it :-P.

It would be nice to see this fixed soon, really appreciate the work that is put to into this!

**Comment by eddie1030** [comment]
wow i didn't expect this to get picked up so quickly, cool!

**Comment by ParitoshBh** [comment]
Sharing a common scenario that's happening with me (and those I am inviting to use RIOT).

I am in the middle of moving away from Whatsapp, Signal, etc. and at the same time asking family members to install and start using RIOT (server end is already up and running). Most of the things work pretty well except for notifications. As mentioned above, majority of users end up not getting any sort of notifications from RIOT on their devices (all are android users) and thereby fallback to using the _traditional_ Whatsapp app for sending message or initiating a call.

Even though there's nothing wrong with RIOT app itself but the fact that its presence is sort of lost (in a sense) and it not being able to deliver notifications reliably garners negative points. Note that there's an option to tinker around with Notification Privacy settings (or something along the lines) but the default setup isn't reliable enough.

Please note that I am not trying to downplay what developers are doing and/or trying to do but this is more of pointing out an adoption issue.

**Comment by manuroe** [comment]
@ganfra, @bmarty , is https://github.com/vector-im/riot-android/pull/2451 the fix for the issue initially described here (` When I switched my phone on, there was no notification of any message on my phone.`)?

**Comment by rjurga** [comment]
This is probably related. I have two phones (Android 8.1.0) with Riot in low detail notifications mode (using Google Cloud Messaging). Neither phone receives notifications at all by default, and then if Riot is started manually, the backlog of notifications appear. This is true for text messages, but also for voice calls, the phone doesn't ring at all unless the app is started manually first.

**Comment by dvb15** [comment]
I have the same on my wife's phone (Meizu m2, Flyme OS v6.2.0.0G, Android 5.1). Even if I left Riot-im running before the phone locks the screen, no notification is played or shown.

Background sync helps, but the messages takes about a minute or two to come... And the voice call too! Heavy load on the servers?

**Comment by bmarty** [comment]
@manuroe yes, I think so

**Comment by bmarty** [comment]
Pb is still there...

**Comment by bmarty** [comment]
To check in next Sprint

**Comment by BloodyIron** [comment]
Yeah, some of my friends are reporting mixed notifications too.

So, Sprint 13 is when this is going to get baked in? That's going to be a new release of the app?

**Comment by BloodyIron** [comment]
Hmm, also may be related to the lazy room sync stuff going on. One of my friends reports he got alerts when the room loaded later or something. It's rather inconsistent and hard to reproduce.

**Comment by bmarty** [comment]
We figure out that signal is sending bad `priority` format

Cf. https://github.com/matrix-org/sygnal/blob/master/sygnal/gcmpushkin.py#L92

According to https://firebase.google.com/docs/cloud-messaging/concept-options it should look like: "android":{"priority":"normal"}. It has maybe change...

That's would explain why notification are not displayed when Riot is killed or after Device Boot

**Comment by eddie1030** [comment]
great job, thanks for figuring out!! Hopefully this fixes it :-)

**Comment by olymk2** [comment]
Agreed I still get multiple reports of people not getting the notifications on riot, unfortunately it generally just works for me seems there is a bigger issue going on here, its actually stopping us using matrix / riot users are using hangouts because the notifications work there :/

**Comment by echto** [comment]
No notification LED on Android on 8.1.0. very annoying.

**Comment by echto** [comment]
Any news?

**Comment by BloodyIron** [comment]
Has this been addressed?

**Comment by echto** [comment]
I installed the latest dev version today to see if things improved and they are actually worse. I receive no notifications at all. I only receive new messages when opening the app. I'm sorry to say this but it appears this app is getting worse and worse. The build I installed today was Build #2447 (Oct 18, 2018 2:14:46 PM).

**Comment by echto** [comment]
The build that works for me, fyi LED notifications do not work with this build, is 8.10-dev (G-b2148).

**Comment by chrissi55** [comment]
The Problem of getting notifications Not reliable under Riot is, as far as i am concerned an old one.

So one day i decided to switch over to Fdroid Version of riot. From that moment i had the settings for background Jobs  and pull notifications by configuring the values for Timeout and delay for checking Interval for new Messages. That did the Trick for all Android Clients in my Family and Friends.

I love the indepedency of Google Push Service by using this alternative. 😁

**Comment by ParitoshBh** [comment]
@chrissi55 How does that impact the battery life? I am assuming it will be worse than using GCM but how bad.

**Comment by chrissi55** [comment]
I can approve, that "battery drain" is higher with Fdroid Version...BUT it is worth it. 
In my case, with Nokia7+ with 3800 mAh and using riot Fdroid Version the whole das... 54% bat in the evening, when taking the phone back to chatging station. 😁
In my opinion, not that bad and the bonus is a reliable notification under matrix 😋

**Comment by BloodyIron** [comment]
@chrissi55 while that may work for you, and is worth noting, that will not work as a solution for the majority. People primarily use Google Play to get their apps, including Riot. This includes myself and my team that uses it. So, we need to hammer out the notification issue when using GCM or whatever is the case.

**Comment by echto** [comment]
+1

On November 2, 2018 6:37:11 AM PDT, BloodyIron <notifications@github.com> wrote:
>@chrissi55 while that may work for you, and is worth noting, that will
>not work as a solution for the majority. People primarily use Google
>Play to get their apps, including Riot. This includes myself and my
>team that uses it. So, we need to hammer out the notification issue
>when using GCM or whatever is the case.
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/vector-im/riot-android/issues/2266#issuecomment-435382416


-- 

BTC: 38RyXXjVuXWqG6bTDfsAqU1KFKXvUbcdcS
ETH: 0x21c15abC275dB22007F7Eed7d8510d1adc6E0195
NEM: NCAQC3HVJ32GM4UUVUJXOLTXCU2GWHHH6EOTBS7C
XRP: rUqCb1HTaNtCHfiKGRcGh6Xz5kipTgBkEU
NEO: AdHHQxwBVvhwbRP1yeDpp9mWzetBmKvh7Z

**Comment by chrissi55** [comment]
...., that will not work as a solution for the majority. ...
ah, sorry 😏

**Comment by bmarty** [comment]
#2706 may improve this

**Comment by echto** [comment]
Latest dev build force closes on me. Clearing cache and data does not resolve.

**Comment by bmarty** [comment]
@echto: oups, is it possible for you to get some log with logcat?

**Comment by echto** [comment]
Here you go.


11-05 05:48:27.570 12505 12505 E AndroidRuntime: FATAL EXCEPTION: main
11-05 05:48:27.570 12505 12505 E AndroidRuntime: Process: im.vector.alpha, PID: 12505
11-05 05:48:27.570 12505 12505 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{im.vector.alpha/im.vector.activity.LoginActivity}: java.lang.IllegalStateException: Default FirebaseApp is not initialized in this process im.vector.alpha. Make sure to call FirebaseApp.initializeApp(Context) first.
11-05 05:48:27.570 12505 12505 E AndroidRuntime:        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2974)
11-05 05:48:27.570 12505 12505 E AndroidRuntime:        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3059)
11-05 05:48:27.570 12505 12505 E AndroidRuntime:        at android.app.ActivityThread.-wrap11(Unknown Source:0)
11-05 05:48:27.570 12505 12505 E AndroidRuntime:        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1724)
11-05 05:48:27.570 12505 12505 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:106)
11-05 05:48:27.570 12505 12505 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:164)
11-05 05:48:27.570 12505 12505 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:7002)
11-05 05:48:27.570 12505 12505 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
11-05 05:48:27.570 12505 12505 E AndroidRuntime:        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:441)
11-05 05:48:27.570 12505 12505 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1408)
11-05 05:48:27.570 12505 12505 E AndroidRuntime: Caused by: java.lang.IllegalStateException: Default FirebaseApp is not initialized in this process im.vector.alpha. Make sure to call FirebaseApp.initializeApp(Context) first.
11-05 05:48:27.570 12505 12505 E AndroidRuntime:        at com.google.firebase.FirebaseApp.getInstance(com.google.firebase:firebase-common@@16.0.2:240)
11-05 05:48:27.570 12505 12505 E AndroidRuntime:        at com.google.firebase.iid.FirebaseInstanceId.getInstance(Unknown Source:1)
11-05 05:48:27.570 12505 12505 E AndroidRuntime:        at im.vector.push.fcm.FcmHelper.ensureFcmTokenIsRetrieved(FcmHelper.java:69)
11-05 05:48:27.570 12505 12505 E AndroidRuntime:        at im.vector.activity.LoginActivity.initUiAndData(LoginActivity.java:370)
11-05 05:48:27.570 12505 12505 E AndroidRuntime:        at im.vector.activity.VectorAppCompatActivity.onCreate(VectorAppCompatActivity.kt:96)
11-05 05:48:27.570 12505 12505 E AndroidRuntime:        at android.app.Activity.performCreate(Activity.java:7258)
11-05 05:48:27.570 12505 12505 E AndroidRuntime:        at android.app.Activity.performCreate(Activity.java:7249)
11-05 05:48:27.570 12505 12505 E AndroidRuntime:        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1221)
11-05 05:48:27.570 12505 12505 E AndroidRuntime:        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2927)
11-05 05:48:27.570 12505 12505 E AndroidRuntime:        ... 9 more

**Comment by bmarty** [comment]
Ok, it's the same understandable error. I will protect the call with a try catch, but FCM will not work for you...

**Comment by echto** [comment]
> I will protect the call with a try catch, but FCM will not work for you...

Ok thank you, but I am a little confused about your message.  FCM will not work?  Do you mean GCM?  And why will it not work?

**Comment by echto** [comment]
Ok, I see from the readme that CGM was changed to FCM.

**Comment by echto** [comment]
Version 0.18.18 (G-b156) from Google Play works.  I received notifications but the LED does not blink.

**Comment by BloodyIron** [comment]
I'm on 0.8.19 (G-b158) and now the app keeps a perpetual notification "Riot.im Listening for Events" and only goes away when I disable "Enable background sync".

So, it seems we've regressed here... what's going on? :/

I'm on Android 7 btw for this device.

**Comment by eddie1030** [comment]
I must say that my experience with the notifications improved a lot. I haven't had the feeling anymore that I saw some message too late. I'm still on android 5 (CM12), and installed Riot w Fdroid.

**Comment by BloodyIron** [comment]
Any word on the permanent notification bit? :/

**Comment by Wykks** [comment]
I'm no longer receiving notification correctly since the version that add a perpetual notification "Riot.im Listening for Events".
I don't get it, the Google Play version should use FCM right ? So why the need of a perpetual notification ?

**Comment by BloodyIron** [comment]
Yeah a good chunk of my friends are getting inconsistent notification activity. Some get it fine, some don't. Some don't even see the setting for background sync.

**Comment by BloodyIron** [comment]
Now I keep getting a notification on android that I've already read on Desktop. This whole notification on android thing is pretty wonky! :(

**Comment by echto** [comment]
Wonky? Yep, you are correct. It's a complete riot!

**Comment by rjurga** [comment]
I commented a long time ago to say that I wasn't receiving notifications, and neither was my wife. It's been a few months now since we're both getting them. Sometimes I even get two notifications for a single message though... Better that none!

**Comment by bmarty** [comment]
@BillCarsonFr can you check this please?

**Comment by BillCarsonFr** [comment]
Hi All,

I am trying to consolidate info from this discussion.
I need to be able to reproduce this in order to investigate more, so I will need your help.

@rjurga You said you are receiving notification now, Are you still using 'Low detail notification' mode?

@BloodyIron Regarding notification for messages already read on desktop. Can you please open a new issue for this feature request? The idea I guess is to delay mobile notifications a bit, and don't send them if message has been read on desktop.

 @Wykks I will investigate on this. Maybe there is an issue while setting up FCM, and the fallback might launch the Fdroid behaviour (this is just speculation, i will check).

@echto Regarding the LED, can you open a new issue on this, 'Add LED light Notification'. If you can add more info on your expectation for lights (for what kind of notifications? colors?) it will be great, thx.
Are you still not receiving any notification? Can you give me more information on your device?

Also, for everyone, can you tell me if you are not receiving notifications at all, or you are receiving only some notifications?

I will also open a new Issue regarding text copy in the notification settings screen. Current text copy is referring to GCM, but GCM is not used anymore (FCM now)

**Comment by BloodyIron** [comment]
@BillCarsonFr the issue for me seems to have happened only once so far. If it happens more, I'll create a new ticket :)

**Comment by BillCarsonFr** [comment]
@ BloodyIron thanks.
As for now I have noticed issues with notifications on devices that have agressive battery optimisation (huawei/honor), and it’s affecting other apps on the device, no notifications.
It can be solved by changing the optimisation settings, and tell the device to not shutdown riot (settings is device/brand/model specific)

**Comment by Wykks** [comment]
No notification for me. (on both onplus 6 and oneplus 5t, but battery optimisation shouldn't be a problem when using FCM right? Other messaging apps doesn't have this issue). Notifications comes when I start the app.

Using riot from play store and encryption enabled if that matters.

**Comment by BillCarsonFr** [comment]
Yes @Wykks , battery optimisation is affecting the app even if using FCM, I can reproduce it and it’s affecting other apps (like tg).
I am happy that it is working for you now, and at the same time sad that I cannot see why it was not working ;)
Can you report a bug by shacking your device so I can investigate your logs? Thx.

**Comment by Wykks** [comment]
Done, but since I reinstalled everything I don't know if you will see anything usefull. I should have done this before, sorry

**Comment by fearedbliss** [comment]
My girlfriend and I are still experiencing this issue, we both have Pixel 3 (Android) with latest stable riot. I've turned off battery optimizations for my app and I sometimes receive the messages and their notifications and sometimes I don't. Sometimes the messages take a free hours to arrive (eventual consistency but that's a long delay and unacceptable for daily use as an instant messenger), sometimes I would see all the messages download when I open it, and recently I've started noticing that if I don't open the app for a long time, it seems that it like it's starting up again (I see the riot multi colored logo loading screen).

**Comment by agherzan** [comment]
I encounter the same issue on two phones. One on oxygen os 9.0.13 (android 9) and one on lineage os 16. Neither show any notifications if the app is closed. Turned off battery optimisation to no result. I'm on riot 0.9.1.

**Comment by echto** [comment]
I believe the problem may be with synapse and not riot-android.

I updated my HS on May 7 and seven days later, on May 14,  notifications ceased occurring for all users on my HS.  Restarting the HS did not correct the issue. Notifications had been working reliable since they were fixed in riot-android.

Restoring to the previous version of synapse corrected the notification issue and notifications have been working reliably again.

**Comment by BloodyIron** [comment]
@echto what version before and after of synapse are we talking about here?

**Comment by geromueller** [comment]
Hi, check out https://github.com/vector-im/riot-android/issues/3162, with 0.99.4 it seems to work again

**Comment by geromueller** [comment]
false alarm. Notifications DO NOT appear if app is not running...

**Comment by echto** [comment]
I downgraded back to version 0.99.1.1.
The version I updated to is 0.99.3.2.  Notifications stopped occuring after 7 days.  Rerverting back to 0.99.1.1 resolved the issue.

**Comment by joram-berger** [comment]
I think my problem is the same as this issue (both Riot apps on Android and for the iphone): 
The phone makes a **sound** when new messages arrive, but the Riot icon has **no indication** of new messages. So if my friends realize that they got some message, they unlock their phone, look at the apps and see no indication what so ever that Riot has new messages. Then they look at Whatsapp and give up searching what it was. That way, new users overlook messages and loose interest in the Riot app.

**Comment by UralZima** [comment]
Hello. As I understood, new versions from F-DROID (microG) don't have background service code anymore, it don't ask for a permission, so that's why it is not working. Because of this, I am using version 0.8.26, which still have background service code... The downgrade was painful, needed to recreate all keys, verify again with everyone, because on android downgrade is not possible, only uninstall and install again.

Now I have another problem... I can't call anymore to IOS users. It seems they upgraded to latest versions, and connection never establishes.
Do new versions have this background service code back? or riot-foss is totally broken and depend from google??? Can I safely upgrade and have notifications working?

**Comment by fearedbliss** [comment]
Not sure why this was closed. The riot android app definitely has flaky notifications. It usually works but sometimes people do message me saying that they haven't received notifications (and this is with my running Synapse 1.4.1 and Riot Android 0.9.8 - the other use is also running Riot Android 0.9.8). I do believe there is usually a combination of issues for users whether it is the Android app not having correct permissions or a bug in synapse, but I don't think all of these issues are resolved. I think this ticket should remain open in order to keep track and perform a wider investigation.

**Comment by BloodyIron** [comment]
> Not sure why this was closed. The riot android app definitely has flaky notifications. It usually works but sometimes people do message me saying that they haven't received notifications (and this is with my running Synapse 1.4.1 and Riot Android 0.9.8 - the other use is also running Riot Android 0.9.8). I do believe there is usually a combination of issues for users whether it is the Android app not having correct permissions or a bug in synapse, but I don't think all of these issues are resolved. I think this ticket should remain open in order to keep track and perform a wider investigation.

Have you guys rage-shake reported the bugs? if not, you should, IMO.

**Comment by ghost** [comment]
This is still an issue. Running Android 10, with 0.9.12 Riot.im app.

Friend sends me a message, and I don't receive any notifications of it. Happens daily, multiple times per day.

**Comment by huguesdk** [comment]
i can confirm this. i have several friends on different homeservers, all with riot.im from google play, and many of them don’t receive notifications (even though all notification debug tests pass). is it possible that the app gets killed somehow and never restarted? most of these people don’t use riot often. someone even tried to exclude riot from battery optimizations, but it didn’t help.

**Comment by UralZima** [comment]
That's why I am still at 0.8.26 version and will never upgrade!!! Because they changed the method and android is killing app anyway. I don't understand, why to not recover old method with persistent notification and background sync? battery is ok with 15-30 sec autosync in settings.
